change/@ni-nimble-components-99393253-07d1-4ec4-902b-41680bddd36e.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Implementation of markdown parser for converting markdown input to rich text content in the viewer",
```

Comment:
>Also in the current build, npm run test fails in Blazor components. Here is the [link](https://github.com/ni/nimble/actions/runs/5423164727/jobs/9862095300?pr=1335#step:17:6569). Is this related to the change done in this PR?

This does appear to be caused by this PR as I don't see the same thing in main. I suspect it has something to do with the rollup config change and ensuring all of nimble's dependencies are correctly being pulled in to all-components.min.js (that bundle is used by Blazor apps).

You can see another version of this failure by opening the [Blazor example app associated with your PR build](https://60e89457a987cf003efc0a5b-tplrazawkl.chromatic.com/blazor-client-app/wwwroot/). It's failing to render all components and shows a bunch of console errors. (You can access this by clicking your build's storybook link and then clicking the "example Blazor app" link on the Getting Started page.
![image](https://github.com/ni/nimble/assets/10500124/cf1464cb-2967-4865-87e4-5d98222d3826)

Some ideas to debug it include inspecting that page or comparing the contents of all-components before and after your change to see if something important is missing.

Reply:
@jattasNI, Thanks for pointing us in the right direction. This helps us to debug the issue and arrive at the root cause of the problem. Below are the findings from the research we did so far and possible solutions for solving the issue.

### Issue:
The `markdown-it` library internally utilizes the core modules of Punycode by importing it with `require('punycode')` ([code here](https://github.com/markdown-it/markdown-it/blob/master/lib/index.js#L14)). However, according to the official Punycode documentation on [GitHub](https://github.com/mathiasbynens/punycode.js/blob/main/README.md#installation), it is recommended to import the userland modules instead (`require('Punycode/')`) of the code module in Node.js. Due to this, the bundle files do not correctly import the Punycode package, resulting in a crash of the Blazor application and so the test files as you mentioned.

Recent issue created in `markdown-it` related to this:
https://github.com/markdown-it/markdown-it/issues/945

### Solution:
During the `npm run build`, the build suggests installing [rollup-plugin-polyfill-node](https://github.com/FredKSchott/rollup-plugin-polyfill-node) as in the below image.
![image](https://github.com/ni/nimble/assets/123377523/f3f0ea50-c718-49c6-90cb-3e94db87c2f9)

By installing the above package and adding the `nodePolyfills()` in the plugins of the `rollup.config.js` file, the missing package is added to the `all-components-bundle.js` file, and the Blazor application runs without any issues. 
Some references for this approach: 
https://github.com/rollup/plugins/pull/51#issuecomment-747489334
https://github.com/rollup/rollup/issues/2881#issuecomment-1048032628

Could you please confirm if it would be okay to proceed with this approach for now?

Reply:
My biggest short term concern is if this is going to add complexity to the Angular build. I know the Angular build configuration is set-up to handle a bunch of weird things like commonjs imports, the process.env global, etc. But I don't know if it will handle this punycode stuff and json imports out of the box.

What I don't want is for Angular clients to need to make their build more complicated to handle Nimble. We'll find out if it's an issue when this change gets pulled into the Angular Directive definition. But yea needing to complicate the Angular build for clients would be a bad idea and we might need to think of other options.

I don't think it needs to block this PR necessarily. I'll try and capture some thoughts on it.

Reply:
Thank you for providing the information. I have incorporated the updates to the Angular directive in this [draft PR](https://github.com/ni/nimble/pull/1346), taking the branch from this PR. After building the bundle with these changes, I tested the components in the example client app and did not encounter any issues with the Angular application. So I am resolving this comment for now and taking up any discussion related to this in the Angular directive PR.